kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLog.java

```diff
@@ -0,0 +1,257 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.streamnative.pulsar.handlers.kop.storage;
+
+import io.netty.buffer.ByteBuf;
+import io.streamnative.pulsar.handlers.kop.KafkaServiceConfiguration;
+import io.streamnative.pulsar.handlers.kop.KafkaTopicManager;
+import io.streamnative.pulsar.handlers.kop.MessagePublishContext;
+import io.streamnative.pulsar.handlers.kop.PendingTopicFutures;
+import io.streamnative.pulsar.handlers.kop.RequestStats;
+import io.streamnative.pulsar.handlers.kop.coordinator.transaction.TransactionCoordinator;
+import io.streamnative.pulsar.handlers.kop.format.EncodeRequest;
+import io.streamnative.pulsar.handlers.kop.format.EncodeResult;
+import io.streamnative.pulsar.handlers.kop.format.EntryFormatter;
+import io.streamnative.pulsar.handlers.kop.format.KafkaMixedEntryFormatter;
+import io.streamnative.pulsar.handlers.kop.utils.MessageMetadataUtils;
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.bookkeeper.mledger.ManagedLedger;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.CorruptRecordException;
+import org.apache.kafka.common.errors.RecordTooLargeException;
+import org.apache.kafka.common.protocol.Errors;
+import org.apache.kafka.common.record.InvalidRecordException;
+import org.apache.kafka.common.record.MemoryRecords;
+import org.apache.kafka.common.record.MutableRecordBatch;
+import org.apache.kafka.common.record.RecordBatch;
+import org.apache.kafka.common.utils.Time;
+import org.apache.pulsar.broker.service.persistent.PersistentTopic;
+import org.apache.pulsar.common.naming.TopicName;
+
+/**
+ * An append-only log for storing messages. Mapping to Kafka Log.scala.
+ */
+@Slf4j
+@AllArgsConstructor
+public class PartitionLog {
+    private KafkaServiceConfiguration kafkaConfig;
+    private Time time;
+    private TopicPartition topicPartition;
+    private String namespacePrefix;
+    private String fullPartitionName;
+    private EntryFormatter entryFormatter;
+    private Optional<TransactionCoordinator> transactionCoordinator;
+
+    public CompletableFuture<Long> appendRecords(final MemoryRecords records,
+                                                 final short version,
+                                                 final AppendRecordsContext appendRecordsContext) {
+        return append(records, version, appendRecordsContext);
+    }
```

Comment:
Should `appendRecords` be replaced by `append`? Nothing more is done in `appendRecords`.

Reply:
Sure. Addressed.